modules/core/04-channel/v2/types/commitment.go

```diff
@@ -12,17 +12,23 @@ import (
 // NOTE: A fixed length preimage is ESSENTIAL to prevent relayers from being able
 // to malleate the packet fields and create a commitment hash that matches the original packet.
 func CommitPacket(packet Packet) []byte {
-	buf := sdk.Uint64ToBigEndian(packet.GetTimeoutTimestamp())
+	var buf []byte
+	timeoutBytes := sdk.Uint64ToBigEndian(packet.GetTimeoutTimestamp())
+	timeoutHash := sha256.Sum256(timeoutBytes)
+	buf = append(buf, timeoutHash[:]...)
 
 	destIDHash := sha256.Sum256([]byte(packet.DestinationChannel))
```

Comment:
Looking at https://github.com/cosmos/ibc/pull/1152/files as reference, is order in which we add to the buf array correct? We do: `[0x2, timeoutHash, destIDHash, appHash]` while in spec `[0x2, destIDHash, timeoutHash, appHash]`